common/include/pcl/pcl_macros.h

```diff
@@ -369,3 +369,14 @@ aligned_free (void* ptr)
 #else
   #define PCL_NODISCARD
 #endif
+
+/**
+ * \brief Macro choose between TYPED_TEST_CASE and PCL_TYPED_TEST_SUITE depending on the GTest version
+ *
```

Comment:
Maybe add a note saying that gtest headers needs to be included before `pcl_macros.h`. Or maybe another else in the if-else ladder using a non-existent macro _INCLUDE_GTEST_BEFORE_PCL_MACROS_H_ so that the error messages show the cause

Reply:
How about a #error? Seems more readable source-code wise.

Reply:
NVM. `#error` will cause unexpected compiler errors in places which doesn't require `gtest.h`

Reply:
I'm questioning if defining these macros here is the best option. Since they are test specific I'm tempted to suggest creating a new file inside the `test` folder:
1- You can include gtest headers without issues.
2- These defines are not exposed by us to consumer projects.

Reply:
So having a separate header which can be included within these test files?

Reply:
Yes. Sorry for being unclear. Create a new header inside the tests folder which all test file will include. A suggestion would be creating another `pcl_macros.h`, this one in `pcl/test/include/pcl/test/pcl_macros.h`. Then all test files would include
```cpp
#include <pcl/test/pcl_macros.h>
```
 It's open to suggestions, especially the name of the file, as it is prone to raise confusion.

Reply:
The new header name seems fine for me.
@kunaltyagi What do you think?

Reply:
`<pcl/test/test_macros.h>` is more constrained, but open enough for future.

Reply:
Done. Waiting for the tests to run.